components/jaggery-core/org.jaggeryjs.jaggery.core/src/main/java/org/jaggeryjs/jaggery/core/manager/JaggeryDeployerManager.java

```diff
@@ -0,0 +1,640 @@
+                errPage.setErrorCode((String) keys);
+                errPage.setLocation((String) arr.get(keys));
+                context.addErrorPage(errPage);
+            }
+        }
+    }
+
+    private static void addFilters(Context ctx, JSONObject jaggeryConfig) {
+        if (jaggeryConfig != null) {
+            JSONArray arrFilters = (JSONArray) jaggeryConfig.get(JaggeryCoreConstants.JaggeryConfigParams.FILTERS);
+            JSONArray arrFilterMappings = (JSONArray) jaggeryConfig
+                    .get(JaggeryCoreConstants.JaggeryConfigParams.FILTER_MAPPINGS);
+
+            if (arrFilters != null) {
+                for (Object filterObj : arrFilters) {
+                    JSONObject filter = (JSONObject) filterObj;
+                    String name = (String) filter.get(JaggeryCoreConstants.JaggeryConfigParams.FILTERS_NAME);
+                    String clazz = (String) filter.get(JaggeryCoreConstants.JaggeryConfigParams.FILTERS_CLASS);
+
+                    FilterDef filterDef = new FilterDef();
+                    filterDef.setFilterName(name);
+                    filterDef.setFilterClass(clazz);
+
+                    JSONArray arrParams = (JSONArray) filter
+                            .get(JaggeryCoreConstants.JaggeryConfigParams.FILTERS_PARAMS);
+                    if (arrParams != null) {
+                        for (Object paramObj : arrParams) {
+                            JSONObject param = (JSONObject) paramObj;
+
+                            String paramName = (String) param
+                                    .get(JaggeryCoreConstants.JaggeryConfigParams.FILTERS_PARAMS_NAME);
+                            String paramValue = (String) param
+                                    .get(JaggeryCoreConstants.JaggeryConfigParams.FILTERS_PARAMS_VALUE);
+
+                            filterDef.addInitParameter(paramName, paramValue);
+                        }
+                    }
+                    ctx.addFilterDef(filterDef);
+                }
+            }
+
+            if (arrFilterMappings != null) {
+                for (Object filterMappingObj : arrFilterMappings) {
+                    JSONObject mapping = (JSONObject) filterMappingObj;
+                    String name = (String) mapping.get(JaggeryCoreConstants.JaggeryConfigParams.FILTER_MAPPINGS_NAME);
+                    String url = (String) mapping.get(JaggeryCoreConstants.JaggeryConfigParams.FILTER_MAPPINGS_URL);
+
+                    FilterMap filterMapping = new FilterMap();
+                    filterMapping.setFilterName(name);
+                    filterMapping.addURLPattern(url);
+                    ctx.addFilterMap(filterMapping);
+                }
+            }
+        }
+    }
+
+    private JSONObject readJaggeryConfig(Context context) {
+        String content = null;
+        if (context.getDocBase().contains(".war")) {
+            try {
+                ZipFile zip = new ZipFile("../webapps/" + context.getDocBase());
```

Comment:
Why this is hard coded? you need to pass the webapp deployment dir path as a parameter for this method, and processJaggeryApp method. And then from jaggery tomcat listener, you need to pass the respective webapp depployment dir path to that method.